src/gmp_tickets_tests.c

```diff
@@ -0,0 +1,112 @@
+/* Copyright (C) 2019 Greenbone Networks GmbH
+ *
+ * SPDX-License-Identifier: GPL-2.0-or-later
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include "gmp_tickets.c"
+
+#include <cgreen/cgreen.h>
+#include <cgreen/mocks.h>
+
+Describe (gmp_tickets);
+BeforeEach (gmp_tickets) {}
+AfterEach (gmp_tickets) {}
+
+/* create_ticket_run */
+
+int
+dummy_client_writer (const char *message, void *data)
+{
+  return 0;
+}
+
+void
+log_event (const char *type, const char *type_name, const char *id,
+           const char *action)
+{
+  return;
+}
+
+gchar *
+ticket_uuid (ticket_t ticket)
+{
+  return g_strdup ("9b5da19e-86b4-11e9-b0d2-28d24461215b");
+}
+
+int
+copy_ticket (const char *comment, const char *ticket_id, ticket_t *new_ticket)
+{
+  mock ();
+  return 0;
+}
+
+int
+create_ticket (const char *comment, const char *result_id,
+               const char *user_id, const char *open_note,
+               ticket_t *ticket)
+{
+  mock ();
+  return 0;
+}
+
+Ensure (gmp_tickets, create_ticket_run_calls_copy_ticket_when_given_copy)
+{
+  gmp_parser_t gmp_parser;
+  GError *error;
+  const gchar *uuid;
+
+  /* Check that create_ticket_run calls only copy_ticket when given COPY. */
+
+  uuid = "9b5da19e-86b4-11e9-b0d2-28d24461215b";
+  gmp_parser.client_writer = dummy_client_writer;
+
+  /* <CREATE_TICKET> */
+  create_ticket_start (&gmp_parser, NULL, NULL);
```

Comment:
Passing NULL pointers for attribute_names and attribute_values in the create_ticket_start(...) call seems to be causing the segfault for me.

Reply:
Oh, I forgot that I patched gvmd-libs: greenbone/gvm-libs/pull/240

Reply:
With the gvm-libs patch it works for me.